### PR TITLE
Add hosts and K8s quickstarts to stateful docs

### PR DESCRIPTION
**Pull request by dedemorton**
Ports hosts quickstart added in https://github.com/elastic/observability-docs/pull/4073 to stateful.

Also makes a few changes to the serverless files to sync them with this PR.

### CODE REVIEW

**Comment by dedemorton** [comment]
@alaudazzi Do you want me to port your changes, too, or do you want to do that?

**Comment by dedemorton** [comment]
@akhileshpok @thomheymann Who is the best person to ask about the privileges required to perform the steps in the hosts quickstart? Also can you please confirm that the hosts quickstart is available in 7.15? I spun up a cluster on cloud last week, and did not see the onboarding UI that I was expecting.

**Comment by alaudazzi** [comment]
@dedemorton please go ahead and port my changes. Thank you and sorry for the late reply.

**Comment by dedemorton** [comment]
I have the same question about K8s monitoring (which I added today): what should we document for required privileges when using stateful?

**Comment by akhileshpok** [comment]
@dedemorton - The hosts and the K8s quickstarts will only become available in stateful in 8.16. They are currently available on serverless. Is your question around required privileges focusing on Elasticsearch, Kibana or something else? Root level privileges are required to install an Elastic Agent for both of these quickstarts.

**Comment by dedemorton** [comment]
>Is your question around required privileges focusing on Elasticsearch, Kibana or something else?

@akhileshpok Yes, I mean are there any Elasticsearch/Kibana privileges that might be required for users who are using a stateful cluster (for serverless, we've documented that the Admin role is required).

**Comment by akhileshpok** [comment]
@thomheymann - Do you have any inputs on the above question related to ES/Kibana privileges? Thanks

**Comment by thomheymann** [comment]
@dedemorton Users need the following privileges to use data onboarding:
- cluster privileges: `['monitor', 'manage_own_api_key']`
- index privileges: `{ names: ['logs-*-*', 'metrics-*-*'], privileges: ['auto_configure', 'create_doc'] }`
- kibana privileges: `{ spaces: ['*'], feature: { fleet: ['all'], fleetv2: ['all'] } }`

The builtin role that has these privileges is the `superuser` role

**Comment by dedemorton** [comment]
Noting here that backport-skip label is set because we don't yet have a branch for 8.16. Everything in master will end up in 8.16 when we create the branch.

**Comment by dedemorton** [comment]
@alaudazzi Can you take a look at my changes in [this commit](https://github.com/elastic/observability-docs/pull/4168/commits/ab558b16182c114a638e878ac88931d29808dbe3) to make sure you agree with the consistency changes that I"ve applied?

File: docs/en/observability/quickstarts/monitor-hosts-with-elastic-agent.asciidoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+preview::[]
+
+In this quickstart guide, you'll learn how to scan your host to detect and collect logs and metrics,
+then navigate to dashboards to further analyze and explore your observability data.
+You'll also learn how to get value out of your observability data.
+
+To scan your host, you'll run an auto-detection script that downloads and installs {agent},
+which is used to collect observability data from the host and send it to Elastic.
+
+The script also generate an {agent} configuration file that you can use with your existing Infrastructure-as-Code tooling.
```
Comment (mdbirnstiehl) [inline docs/en/observability/quickstarts/monitor-hosts-with-elastic-agent.asciidoc:13]:
```suggestion
The script also generates an {agent} configuration file that you can use with your existing Infrastructure-as-Code tooling.
```

File: docs/en/observability/quickstarts/monitor-hosts-with-elastic-agent.asciidoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+. Select **Collect and analyze logs**, and then select **Auto-detect logs and metrics**.
+. Copy the command that's shown. For example:
++
+[role="screenshot"]
+image::images/quickstart-autodetection-command.png[Quick start showing command for running auto-detection]
++
+You'll run this command to download the auto-detection script and scan your system for observability data.
+. Open a terminal on the host you want to scan, and run the command.
+. Review the list of log files:
+    * Enter `Y` to ingest all the log files listed
```
Comment (mdbirnstiehl) [inline docs/en/observability/quickstarts/monitor-hosts-with-elastic-agent.asciidoc:51]:
```suggestion
    * Enter `Y` to ingest all the log files listed.
```
For punctuation consistency with the next bullet.

File: docs/en/observability/images/quickstart-k8s-overview.png
Comment (mdbirnstiehl) [inline docs/en/observability/images/quickstart-k8s-overview.png:1]: do we want to show the navigation in this graphic?

**Review by mdbirnstiehl — COMMENTED** [review]
A couple of small things but otherwise LGTM!

**Review by mdbirnstiehl — DISMISSED** [review]

File: docs/en/observability/images/quickstart-k8s-overview.png
Comment (dedemorton) [reply docs/en/observability/images/quickstart-k8s-overview.png:1]: Nope, I'll crop it because that's the serverless navigation.

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-fa312ctrt-elastic-dev.vercel.app

* Built with commit c0bfdfaca1560bb3429cf164f4805a68f453f37d

> Issues? Visit #next-docs in Slack

**Comment by dedemorton** [comment]
@akhileshpok I'm going to merge this because it's blocking another PR I have in progress. If you have any concerns with the changes here, just open a docs issue, and we will address your concerns. Thanks!

**Pull request merged**